src/Adapter/Product/Validate/ProductValidator.php

```diff
@@ -39,6 +39,22 @@
  */
 class ProductValidator extends AbstractObjectModelValidator
 {
+    /**
+     * This method is specific for product creation only.
+     *
+     * @param Product $product
+     *
+     * @throws CoreException
+     */
+    public function validateCreation(Product $product): void
+    {
+        $this->validateProductLocalizedProperty(
```

Comment:
Just one question, `is_virtual` doesn't need validation?

Reply:
late answer (answered in slack). But for the record, it is a boolean & we make sure it is using php types, nothing more to validate here

Reply:
@matks, @jolelievre Im coming back to this question :smile: 
So I thought about it and Im abit confused.
At first I thought we don't need to validate those fields which in modern php is supposed to be handled by strict typing (bool, int validations).
**BUT**
Validators accepts legacyObject model which doesn't have strict types, so in theory anyone using validator (without using our commands) will be able to pass invalid values
So maybe it is right to validate all the properties after all? :thinking:

Reply:
I guess you're right it would be safer to validate all the fields

Reply:
Always better to double check!